examples/pybullet/pybullet.c

```diff
@@ -6947,6 +6953,147 @@ static PyObject* pybullet_calculateInverseDynamics(PyObject* self,
+		PyObject* objAccelerations;
+		int physicsClientId = 0;
+		b3PhysicsClientHandle sm = 0;
+		static char* kwlist[] = {"bodyUniqueId", "linkIndex", "localPosition",
+								 "objPositions", "objVelocities",
+								 "objAccelerations", "physicsClientId", NULL};
+		if (!PyArg_ParseTupleAndKeywords(args, keywds, "iiOOOO|i", kwlist,
+				&bodyUniqueId, &linkIndex, &localPosition, &objPositions,
+				&objVelocities, &objAccelerations, &physicsClientId))
+		{
+			return NULL;
+		}
+		sm = getPhysicsClient(physicsClientId);
+		if (sm == 0)
+		{
+			PyErr_SetString(SpamError, "Not connected to physics server.");
+			return NULL;
+		}
+
+		{
+			int szLoPos = PySequence_Size(localPosition);
+			int szObPos = PySequence_Size(objPositions);
+			int szObVel = PySequence_Size(objVelocities);
+			int szObAcc = PySequence_Size(objAccelerations);
+			int numJoints = b3GetNumJoints(sm, bodyUniqueId);
+			if (numJoints && (szLoPos == 3) && (szObPos == numJoints) && 
+				(szObVel == numJoints) && (szObAcc == numJoints))
+			{
+				int byteSizeJoints = sizeof(double) * numJoints;
+				int byteSizeVec3 =  sizeof(double) * 3;
+				int i;
+				PyObject* pyResultList = PyTuple_New(2);
+				double* localPoint = (double*)malloc(byteSizeVec3);
+				double* jointPositions = (double*)malloc(byteSizeJoints);
+				double* jointVelocities = (double*)malloc(byteSizeJoints);
+				double* jointAccelerations = (double*)malloc(byteSizeJoints);
+				double* linearJacobian = (double*)malloc(3 * byteSizeJoints);
+				double* angularJacobian = (double*)malloc(3 * byteSizeJoints);
+
+				pybullet_internalSetVectord(localPosition, localPoint);
+				for (i = 0; i < numJoints; i++)
+				{
+					jointPositions[i] =
+						pybullet_internalGetFloatFromSequence(objPositions, i);
+					jointVelocities[i] =
+						pybullet_internalGetFloatFromSequence(objVelocities, i);
+					jointAccelerations[i] =
+						pybullet_internalGetFloatFromSequence(objAccelerations, i);
+				}
+				{
+					b3SharedMemoryStatusHandle statusHandle;					
+					b3SharedMemoryCommandHandle commandHandle =
+						b3CalculateJacobianCommandInit(sm, bodyUniqueId,
+							linkIndex, localPoint, jointPositions,
+							jointVelocities, jointAccelerations);
+					statusHandle = b3SubmitClientCommandAndWaitStatus(sm, commandHandle);
+					int statusType = b3GetStatusType(statusHandle);
+					if (statusType == CMD_CALCULATED_JACOBIAN_COMPLETED)
+					{
+						int dofCount;
+						b3GetStatusJacobian(statusHandle, &dofCount, NULL, NULL);
+						if (dofCount)
+						{
+							int byteSizeDofCount = sizeof(double) * dofCount;
+							double* linearJacobian = (double*)malloc(3 * byteSizeDofCount);
+							double* angularJacobian = (double*)malloc(3 * byteSizeDofCount);
+							b3GetStatusJacobian(statusHandle,
+												NULL,
+												linearJacobian,
+												angularJacobian);
+							if (linearJacobian)
+							{
+								PyObject* pymat = PyTuple_New(3);
+								for (int r = 0; r < 3; ++r) {
```

Comment:
you can move the 'int r' out of the loop, to satisfy CI?

Reply:
GCC 5 enables C99 features by default. Is it possible to upgrade to at least that instead?

Reply:
This is Visual Studio compiling C code, simple fix.